src/Raven.Studio/typescript/viewmodels/resources/widgets/clusterOverviewWidget.ts

```diff
@@ -0,0 +1,288 @@
+                    onValue(moment.utc(nodeItem.startTime()), nodeItem.startTime());
+                }
+            });
+        
+        this.enableSyncUpdates();
+
+        for (let ws of this.controller.getConnectedLiveClients()) {
+            this.onClientConnected(ws);
+        }
+
+        this.initTooltips();
+    }
+
+    private prepareGridData(): JQueryPromise<pagedResult<nodeStatsItem>> {
+        let items = this.nodeStats();
+
+        this.applyPerDatabaseStripes(items);
+        
+        return $.when({
+            totalResultCount: items.length,
+            items
+        });
+    }
+
+    protected prepareColumns(): virtualColumn[] {
+        const grid = this.gridController();
+        
+        return [
+            new nodeTagColumn<nodeStatsItem>(grid, item => this.prepareUrl(item, "Cluster Dashboard")),
+
+            new iconsPlusTextColumn<nodeStatsItem>(grid, item => item.disconnected() ? "Trying to connect.." : (item.nodeType() ? this.nodeTypeDataForHtml(item.nodeType()) : ""), "Type", "15%"),
+
+            new iconsPlusTextColumn<nodeStatsItem>(grid, item => item.nodeState() ? this.nodeStateDataForHtml(item.nodeState()) : "", "State", "15%"),
+
+            new textColumn<nodeStatsItem>(grid, item => item.formattedUpTime() || "", "Up time", "15%"),
+
+            new textColumn<nodeStatsItem>(grid, item => item.formattedStartTime() || "", "Start time", "20%"),
+            
+            new actionColumn<nodeStatsItem>(grid, item => router.navigate(item.nodeUrl()), "URL", item => item.nodeUrl() || "" , "20px",
+                {
+                    title: () => 'Go to URL'
+                }),
+        ];
+    }
+    
+    private nodeTypeDataForHtml(type: clusterNodeType): iconPlusText[] {
+        let iconClass;
+        
+        switch (type) {
+            case "Member": iconClass = "icon-cluster-member";
+            break;
+            case "Promotable": iconClass = "icon-cluster-promotable";
+            break;
+            case "Watcher": iconClass= "icon-cluster-watcher";
+            break;
+            default:
+                console.warn("Invalid node type: " + type);
+            break;
+        }
+        return [{
+            title: "",
+            text: type,
+            iconClass: iconClass,
+            textClass: ""
+        }];
+    }
+
+    private nodeStateDataForHtml(state: Raven.Client.ServerWide.RachisState): iconPlusText[] {
+        let iconClass;
+
+        switch (state) {
+            case "Leader":
+                iconClass = "icon-node-leader";
+                break;
+
+            case "Passive":
+            case "Candidate":
+            case "Follower":
+            case "LeaderElect":
+                iconClass = "";
+                break;
+
+            default:
+                console.warn("Invalid node state: " + state);
+                break;
+        }
+
+        return [{
+            title: "",
```

Comment:
let's make title and text class optional and handle that correctly in iconsPlusTextColumn

Reply:
done